### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by avminon**
Task #54210 - Auth views and routes

### CODE REVIEW

File: app/Http/Controllers/HomeController.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+class HomeController extends Controller
+{
+    /**
+     * Create a new controller instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        $this->middleware('auth');
```
Comment (dinhhoanglong91) [inline app/Http/Controllers/HomeController.php]: Please set middleware in router

File: resources/views/auth/login.blade.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
+@extends('layouts.app')
+
+@section('content')
+<div class="container">
+    <div class="row">
+        <div class="col-md-8 col-md-offset-2">
+            <div class="panel panel-default">
+                <div class="panel-heading">Login</div>
+                <div class="panel-body">
+                    <form class="form-horizontal" role="form" method="POST" action="{{ url('/login') }}">
```
Comment (dinhhoanglong91) [inline resources/views/auth/login.blade.php]: Form::open()

File: resources/views/auth/login.blade.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                        {!! csrf_field() !!}
+
+                        <div class="form-group{{ $errors->has('email') ? ' has-error' : '' }}">
+                            <label class="col-md-4 control-label">E-Mail Address</label>
```
Comment (dinhhoanglong91) [inline resources/views/auth/login.blade.php]: Use trans
https://github.com/framgia/tms_47/pull/11#discussion_r53422207

File: resources/views/auth/login.blade.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+                            <div class="col-md-6">
+                                <input type="email" class="form-control" name="email" value="{{ old('email') }}">
+
+                                @if ($errors->has('email'))
+                                    <span class="help-block">
+                                        <strong>{{ $errors->first('email') }}</strong>
+                                    </span>
+                                @endif
```
Comment (nguyenthanhtung88) [inline resources/views/auth/login.blade.php]: This `if` block format i saw you used in multiple location.
Why don't you make an element (partial view) and include in specific view.

File: resources/views/auth/login.blade.php [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+                <div class="panel-heading">Login</div>
+                <div class="panel-body">
+                    <form class="form-horizontal" role="form" method="POST" action="{{ url('/login') }}">
+                        {!! csrf_field() !!}
+
+                        <div class="form-group{{ $errors->has('email') ? ' has-error' : '' }}">
+                            <label class="col-md-4 control-label">E-Mail Address</label>
+
+                            <div class="col-md-6">
+                                <input type="email" class="form-control" name="email" value="{{ old('email') }}">
```
Comment (nguyenthanhtung88) [inline resources/views/auth/login.blade.php]: Try to use Form Helper in your view

File: resources/views/auth/login.blade.php [hunk 6]
```diff
@@ -0,0 +1,81 @@
+@extends('layouts.app')
+
+@section('content')
+<div class="container">
+    <div class="row">
+        <div class="col-md-8 col-md-offset-2">
+            <div class="panel panel-default">
+                <div class="panel-heading">Login</div>
```
Comment (nguyenthanhtung88) [inline resources/views/auth/login.blade.php]: Use `trans()` for static text, please check another static text you used

File: resources/views/auth/register.blade.php [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+                            <div class="col-md-6">
+                                {!! Form::text('name', '', [
+                                        'required' => 'required',
+                                        'placeholder' => 'Enter name here',
+                                        'class' => 'form-control',
+                                        'value' => "{{ old('name') }}"
+                                    ])
+                                !!}
+
+                                @include('errors.name')
```
Comment (nguyenthanhtung88) [inline resources/views/auth/register.blade.php]: Oops, in my suggestion, my idea is created a partial like `errors.user_register_input`, and pass field name param.
For example, i have to call `@include('errors.user_register_input', ['field' => 'name'])` and we don't have to create multiple partial for doing the same thing, they only have `field name` difference.

I saw you create `errors.name`, `errors.email`, ... that's not necessary!

File: resources/views/auth/passwords/reset.blade.php [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+                            <label class="col-md-4 control-label">E-Mail Address</label>
+
+                            <div class="col-md-6">
+                                <input type="email" class="form-control" name="email" value="{{ $email or old('email') }}">
+
+                                @if ($errors->has('email'))
+                                    <span class="help-block">
+                                        <strong>{{ $errors->first('email') }}</strong>
+                                    </span>
+                                @endif
```
Comment (nguyenthanhtung88) [inline resources/views/auth/passwords/reset.blade.php]: You had partial view defined right? why didn't you make changes in this location or others?

File: resources/lang/en/common.php [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+        'page_title' => 'Framgia Task Management System',
+        'framgia' => 'Framgia',
+        'dashboard' => 'Dashboard',
+        'home' => 'Home',
+        'toggleNav' => 'Toggle Navigation',
+        'name' => 'Name',
+        'email' => 'Email',
+        'password' => 'Password',
+        'confirmPassword' => 'Confirm Password',
+        'login' => '<i class="fa fa-btn fa-sign-in"></i>Login',
```
Comment (nguyenthanhtung88) [inline resources/lang/en/common.php]: Don't write html code in language file, `Login` is enough, put html code in view

File: resources/views/errors/email.blade.php [hunk 10]
```diff
@@ -0,0 +1,5 @@
+@if ($errors->has('email'))
```
Comment (nguyenthanhtung88) [inline resources/views/errors/email.blade.php]: Do you need this partial view?

File: resources/views/layouts/app.blade.php [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+    <!-- Fonts -->
+    <link href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.4.0/css/font-awesome.min.css" rel='stylesheet' type='text/css'>
+    <link href="https://fonts.googleapis.com/css?family=Lato:100,300,400,700" rel='stylesheet' type='text/css'>
+
+    <!-- Styles -->
+    <link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css" rel="stylesheet">
+    {{-- <link href="{{ elixir('css/app.css') }}" rel="stylesheet"> --}}
+
+    <style>
+        body {
```
Comment (nguyenthanhtung88) [inline resources/views/layouts/app.blade.php]: Don't write style define in your view, make css file for this

File: resources/views/welcome.blade.php [hunk 12]
```diff
@@ -1,45 +1,17 @@
...
-        <title>Laravel</title>
+@extends('layouts.app')
 
-        <link href="https://fonts.googleapis.com/css?family=Lato:100" rel="stylesheet" type="text/css">
+@section('content')
+<div class="container">
+    <div class="row">
+        <div class="col-md-10 col-md-offset-1">
+            <div class="panel panel-default">
+                <div class="panel-heading">Welcome</div>
```
Comment (nguyenthanhtung88) [inline resources/views/welcome.blade.php]: Use `trans()`

File: resources/views/errors/email.blade.php (hunk 10 above)
Comment (avminon) [inline resources/views/errors/email.blade.php]: I thought this was deleted already in my previous PR sir. Sorry will update it now

**Comment by nguyenthanhtung88** [comment]
OK

File: resources/views/auth/passwords/email.blade.php [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+                        </div>
+                    @endif
+
+                    {!! Form::open(['method' => 'post'], [
+                            'class' => 'form-horizontal',
+                            'role' => 'form',
+                            'action' => "url('/password/email')",
+                        ])
+                    !!}
+                        {!! csrf_field() !!}
```
Comment (dinhhoanglong91) [inline resources/views/auth/passwords/email.blade.php]: `Form::open` automatically generate CSRF

File: resources/views/auth/passwords/reset.blade.php [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+                </div>
+
+                <div class="panel-body">
+                    {!! Form::open(['method' => 'post'], [
+                            'class' => 'form-horizontal',
+                            'role' => 'form',
+                            'action' => "url('/password/reset')",
+                        ])
+                    !!}
+                        {!! csrf_field() !!}
```
Comment (dinhhoanglong91) [inline resources/views/auth/passwords/reset.blade.php]: remove it

File: resources/views/auth/register.blade.php [hunk 15]
```diff
@@ -0,0 +1,99 @@
...
+            <div class="panel panel-default">
+                <div class="panel-heading"><i class="fa fa-btn fa-book"></i>{!! trans('common.main.register') !!}</div>
+                <div class="panel-body">
+                    {!! Form::open(['method' => 'post'], [
+                            'class' => 'form-horizontal',
+                            'role' => 'form',
+                            'action' => "url('/login')",
+                        ])
+                    !!}
+                         {!! csrf_field() !!}
```
Comment (dinhhoanglong91) [inline resources/views/auth/register.blade.php]: remove it

File: resources/views/auth/login.blade.php [hunk 16]
```diff
@@ -0,0 +1,80 @@
+@extends('layouts.app')
+
+@section('content')
+<div class="container">
+    <div class="row">
+        <div class="col-md-8 col-md-offset-2">
+            <div class="panel panel-default">
+                <div class="panel-heading"><i class="fa fa-btn fa-sign-in"></i>{!! trans('common.main.login') !!}</div>
```
Comment (wataridori) [inline resources/views/auth/login.blade.php]: In your files, you are using both `{!! trans() !!}` and `{{ trans() }}`. 
Please use only one way of writing. I suggest using `{{ trans() }}`.

**Comment by avminon** [comment]
Accidentally force pushed into closing while fixing conflicts

File: resources/views/auth/register.blade.php [hunk 17]
```diff
@@ -0,0 +1,97 @@
...
+
+                        <div class="form-group{{ $errors->has('email') ? ' has-error' : '' }}">
+                            {!! Form::label('email', trans('common.main.email'), ['class' => 'col-md-4']); !!}
+
+                            <div class="col-md-6">
+                                {!! Form::email('email', '', [
+                                        'required' => 'required',
+                                        'placeholder' => 'Enter email here',
+                                        'class' => 'form-control',
+                                        'value' => "{{ old('email') }}"
```
Comment (dinhhoanglong91) [inline resources/views/auth/register.blade.php]:
```php
'value' => old('email')
```

File: resources/views/auth/register.blade.php [hunk 18]
```diff
@@ -0,0 +1,97 @@
...
+                        </div>
+
+
+                        <div class="form-group{{ $errors->has('email') ? ' has-error' : '' }}">
+                            {!! Form::label('email', trans('common.main.email'), ['class' => 'col-md-4']); !!}
+
+                            <div class="col-md-6">
+                                {!! Form::email('email', '', [
+                                        'required' => 'required',
+                                        'placeholder' => 'Enter email here',
```
Comment (dinhhoanglong91) [inline resources/views/auth/register.blade.php]: trans

File: resources/views/auth/register.blade.php [hunk 19]
```diff
@@ -0,0 +1,97 @@
...
+                    !!}
+                         <div class="form-group{{ $errors->has('name') ? ' has-error' : '' }}">
+                            {!! Form::label('name', trans('common.main.name'), ['class' => 'col-md-4']); !!}
+
+                            <div class="col-md-6">
+                                {!! Form::text('name', '', [
+                                        'required' => 'required',
+                                        'placeholder' => 'Enter name here',
+                                        'class' => 'form-control',
+                                        'value' => "{{ old('name') }}"
```
Comment (dinhhoanglong91) [inline resources/views/auth/register.blade.php]: https://github.com/framgia/tms_47/pull/10/files#r53598194

Comment (dinhhoanglong91) [inline resources/views/auth/register.blade.php]:
```php
'value' => old('email')
```

File: resources/views/auth/login.blade.php [hunk 20]
```diff
@@ -0,0 +1,80 @@
...
+                    !!}
+                        <div class="form-group{{ $errors->has('email') ? ' has-error' : '' }}">
+                            {!! Form::label('email', trans('common.main.email'), ['class' => 'col-md-4']); !!}
+
+                            <div class="col-md-6">
+                                {!! Form::email('email', '', [
+                                        'required' => 'required',
+                                        'placeholder' => 'Enter email here',
+                                        'class' => 'form-control',
+                                        'value' => "{{ old('email') }}"
```
Comment (dinhhoanglong91) [inline resources/views/auth/login.blade.php]:
```php
'value' => old('email')
```

File: resources/views/auth/passwords/email.blade.php [hunk 21]
```diff
@@ -0,0 +1,60 @@
...
+                                    'class' => 'col-md-4 control-label'
+                                ]);
+                            !!}
+
+                            <div class="col-md-6">
+                                {!! Form::email('email', '', [
+                                        'required' => 'required',
+                                        'placeholder' => 'Enter email here',
+                                        'class' => 'form-control',
+                                        'value' => "{{ old('email') }}"
```
Comment (dinhhoanglong91) [inline resources/views/auth/passwords/email.blade.php]:
```php
'value' => old('email')
```

File: resources/views/auth/passwords/reset.blade.php [hunk 22]
```diff
@@ -0,0 +1,91 @@
...
+                                    'class' => 'col-md-4 control-label'
+                                ]);
+                            !!}
+
+                            <div class="col-md-6">
+                                {!! Form::email('email', '', [
+                                        'required' => 'required',
+                                        'placeholder' => 'Enter email here',
+                                        'class' => 'form-control',
+                                        'value' => "{{ old('email') }}"
```
Comment (dinhhoanglong91) [inline resources/views/auth/passwords/reset.blade.php]:
```php
'value' => old('email')
```

**Comment by dinhhoanglong91** [comment]
ok

**Pull request merged**